### PR TITLE
Fix JS tests failing on master

### PR DESCRIPTION
**Pull request by timja**
Apparently JS unit tests didn't cause the build to fail when `maven.test.failure.ignore` was set see https://github.com/jenkinsci/jenkins/pull/5031#issuecomment-716205257

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
Well, it definitely makes sense to migrate tests to divs

**Comment by jglick** [comment]
What test failure? URL?

**Comment by jglick** [comment]
See also #5032.

**Comment by timja** [comment]
> What test failure? URL?

looks like they might not be run in CI, @StefanSpieker raised it in https://github.com/jenkinsci/jenkins/pull/5006

**Comment by StefanSpieker** [comment]
It is also in CI build, but the build is not failing. See for example: [PR-5030 build log](https://ci.jenkins.io/blue/rest/organizations/jenkins/pipelines/Core/pipelines/jenkins/branches/PR-5030/runs/2/nodes/51/steps/57/log/?start=0)
Just search for the `Tests:       6 failed, 14 passed, 20 total`

**Review by StefanSpieker — APPROVED** [review]

**Comment by jglick** [comment]
https://ci.jenkins.io/job/Core/job/jenkins/job/master/2235/execution/node/45/log/?consoleFull shows the failure, but I think because we are passing `-Dmaven.test.failure.ignore` it is treated as nonfatal? So the question is how to report this in `junit`-compatible format.

**Comment by timja** [comment]
Yes just found that, looking in to it

**Comment by jglick** [comment]
https://github.com/eirslett/frontend-maven-plugin/blob/cdc515af830fb0be5f0a1710c538b7b950fef1f9/frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/AbstractFrontendMojo.java#L96-L101

**Comment by timja** [comment]
Also FTR this has tripped us up in a release before, javascript linting was failing during release but not in CI, same cause I assume.

**Comment by timja** [comment]
RFE for it:
https://github.com/eirslett/frontend-maven-plugin/issues/679

I guess to fix this we need to move running it outside of maven.

**Comment by jglick** [comment]
https://github.com/eirslett/frontend-maven-plugin/pull/564

Is some file created when tests _pass_ that we can assert the existence of?

**Comment by timja** [comment]
I think we can configure a test reporter

**Comment by jglick** [comment]
With `warnings-ng` you mean? That is one option I guess.

**Comment by jglick** [comment]
Search for `com.github.eirslett.maven.plugins.frontend.lib.TaskRunnerException: `?

**Comment by timja** [comment]
no no just a standard javascript one that emits junit results, almost done, testing locally.

**Comment by timja** [comment]
I've temp reverted the js unit test fix just to see it fail in CI

**Comment by timja** [comment]
Failed correctly in https://ci.jenkins.io/blue/organizations/jenkins/Core%2Fjenkins/detail/PR-5031/2/tests/

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
expediting so that we don't get a failed release tomorrow by forgetting to merge this...

File: war/.gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 work
 /rebel.xml
+junit.xml
```
Comment (jglick) [inline war/.gitignore:3]: Would prefer to place it under `target/`.

**Review by jglick — COMMENTED** [review]
Thanks for fixing this!

**Pull request merged**